### PR TITLE
Existing bugs

### PR DESCRIPTION
**Pull request by Noor-mommadhi**
ISSUES FIXES: #7916

1)Issue 9:For each ticket, the Ticket conversion/description is shown twice.

2)Issue 15:Data is not vanishing on create organization page. click on create, input the data to the organization name, and click on close

3)Issue 47:While adding the language for the ISO-CODE, throwing the wrong alert message for the number of characters.

4)Issue 48:In cron settings, click on the link redirecting to a blank page.

5)Issue 51: In the knowledge base setting textbox. Suppose we don’t input the data to the required field and submit it. By default, it's picking the value as ‘0’.

6)Issue 20:No option available to remove the due date in ticket creation.

7)Issue 10:By default ticket due date is picked as the ticket created date. (if the ticket due date is selected on a different date. But, still picked the same date.

### CODE REVIEW

**Review by sowmiyasubramaniyan — APPROVED** [review]

File: app/Http/Controllers/Agent/helpdesk/TicketController.php [hunk 1]
```diff
@@ -2344,15 +2344,23 @@ public function getTooltip(Request $request)
...
+                $tooltip .= '<b>'.$thread->user->user_name.' ('.$thread->poster.')</b></br>'
+                    .$thread->purify().'<br><hr>';
+            }
+
+            $tooltip .= 'This ticket has '.$numThreads.' threads.';
 
+            return $tooltip;
+        }
+
+        //  return '';
```
Comment (DhanyaChengappa) [inline app/Http/Controllers/Agent/helpdesk/TicketController.php:2374]: remove this line if not required

**Review by RafficMohammed — CHANGES_REQUESTED** [review]
@Noor-mommadhi U added too many unwanted files. Please remove it

**Review by SarfrazS — CHANGES_REQUESTED** [review]
Issue 1:- while adding language with valid inputs getting below error, but language added.
image
image

Rest all the below issues are fixed and working fine.
1)Issue 9:For each ticket, the Ticket conversion/description is shown twice.

2)Issue 15:Data is not vanishing on create organization page. click on create, input the data to the organization name, and click on close

3)Issue 47:While adding the language for the ISO-CODE, throwing the wrong alert message for the number of characters.

4)Issue 48:In cron settings, click on the link redirecting to a blank page.

5)Issue 51: In the knowledge base setting textbox. Suppose we don’t input the data to the required field and submit it. By default, it's picking the value as ‘0’.

6)Issue 20:No option available to remove the due date in ticket creation.

7)Issue 10:By default ticket due date is picked as the ticket created date. (if the ticket due date is selected on a different date. But, still picked the same date.

**Comment by Umasri-V** [comment]
found new issues https://docs.google.com/document/d/1LvC-KUd3fTn8OIcxFzcTSpL0D0Qnci5RznFZQr3GeP0/edit

File: app/Http/Controllers/Agent/helpdesk/TicketController.php [hunk 2]
```diff
@@ -632,7 +630,7 @@ public function checkMobile($mobile)
      *
      * @return type bool
      */
-    public function create_user($emailadd, $username, $subject, $body, $phone, $phonecode, $mobile_number, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $from_data, $auto_response, $status)
+    public function create_user($emailadd, $username, $subject, $body, $phone, $phonecode, $mobile_number, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $from_data, $auto_response, $status, $duedate)
     {
```
Comment (RafficMohammed) [inline app/Http/Controllers/Agent/helpdesk/TicketController.php:634]: add $duedate = null in arguments

File: app/Http/Controllers/Agent/helpdesk/TicketController.php [hunk 3]
```diff
@@ -914,7 +912,7 @@ public function default_priority()
      *
      * @return type string
      */
-    public function check_ticket($user_id, $subject, $body, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $form_data, $status)
+    public function check_ticket($user_id, $subject, $body, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $form_data, $status, $duedate)
```
Comment (RafficMohammed) [inline app/Http/Controllers/Agent/helpdesk/TicketController.php:915]: Here too

File: app/Http/Controllers/Agent/helpdesk/TicketController.php [hunk 4]
```diff
@@ -990,7 +988,7 @@ public function check_ticket($user_id, $subject, $body, $helptopic, $sla, $prior
      *
      * @return type string
      */
-    public function createTicket($user_id, $subject, $body, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $form_data, $status)
+    public function createTicket($user_id, $subject, $body, $helptopic, $sla, $priority, $source, $headers, $dept, $assignto, $form_data, $status, $duedate)
```
Comment (RafficMohammed) [inline app/Http/Controllers/Agent/helpdesk/TicketController.php:991]: Here too

**Comment by Sakthi002** [comment]
> @Noor-mommadhi U added too many unwanted files. Please remove it

Auto Generated files sir

File: app/Http/Controllers/Agent/helpdesk/TicketController.php [hunk 5]
```diff
@@ -1038,9 +1036,21 @@ public function createTicket($user_id, $subject, $body, $helptopic, $sla, $prior
         $ticket->save();
 
         $sla_plan = Sla_plan::where('id', '=', $sla)->first();
-        $ovdate = $ticket->created_at;
-        $new_date = date_add($ovdate, date_interval_create_from_date_string($sla_plan->grace_period));
-        $ticket->duedate = $new_date;
+        if ($duedate) {
+            $duedate_datetime = \DateTime::createFromFormat('d/m/Y', $duedate);
```
Comment (RafficMohammed) [inline app/Http/Controllers/Agent/helpdesk/TicketController.php:1040]: U can use Carbon facades instead of Datetime. Why are u using this ?

**Comment by RafficMohammed** [comment]
@Noor-mommadhi When you r adding any new argument, you must be very careful. @SarfrazS Check ticket fetch functionality here

**Review by SarfrazS — APPROVED** [review]
Pointed issues are fixed.

**Review by RafficMohammed — APPROVED** [review]

**Comment by RafficMohammed** [comment]
@Noor-mommadhi Squash commits

**Pull request merged**